src/app/components/pages/calendar/calendar.component.ts

```diff
@@ -0,0 +1,138 @@
+import {Component, EventEmitter, Input, OnInit, Output} from '@angular/core';
+import {Day, Period, PeriodOfDay} from '../../../models/calendar';
+import {DatePipe} from '@angular/common';
+import {reserveSlots} from '@angular/core/src/render3/instructions';
+
+@Component({
+  selector: 'app-calendar',
+  templateUrl: './calendar.component.html',
+  styleUrls: ['./calendar.component.scss'],
+  providers: [DatePipe],
+})
+export class CalendarComponent implements OnInit {
+  _timeSlots: Period[] = [];
+  @Input()
+  get timeSlots() {
+    return this._timeSlots;
+  }
+
+  set timeSlots(time_slots) {
+    if (time_slots.length > 0) {
+      this._timeSlots = time_slots;
+      this.refreshDaysWeek();
+    }
+  }
+
+  @Output() period_emiiter = new EventEmitter<Period>();
+
+  days_of_Week_titles: Day[];
+
+  constructor(public datepipe: DatePipe) {
+  }
+
+  ngOnInit() {
+    this.days_of_Week_titles = this.startAndEndDaysOfWeek();
+  }
+
+  getEndStartWeek() {
+    if (this.days_of_Week_titles.length < 2) {
+      return null;
+    }
+    return [
+      this.datepipe.transform(
+        this.days_of_Week_titles[0].date,
+        'dd/MM/yyyy'
+      ),
+      this.datepipe.transform(
+        this.days_of_Week_titles[6].date,
+        'dd/MM/yyyy'
+      )
+    ];
+  }
+
+  refreshDaysWeek() {
+    for (const day of this.days_of_Week_titles) {
+      const periods_tmp: Period[] = [null, null, null];
+      for (const period of this.timeSlots) {
+        const dateweek = this.getDateWithoutHours(day.date);
+        const star_date_timeslot = this.getDateWithoutHours(period.start);
+        const end_date_timeslot = this.getDateWithoutHours(period.end);
+
+        if ((dateweek >= star_date_timeslot) && (dateweek <= end_date_timeslot)) {
+          if (period.period_day) {
+            if (period.period_day === PeriodOfDay.am) {
+              periods_tmp[0] = period;
+            }
+            if (period.period_day === PeriodOfDay.pm) {
+              periods_tmp[1] = period;
+            }
+            if (period.period_day === PeriodOfDay.night) {
+              periods_tmp[2] = period;
+            }
+          }
+        }
+      }
+      day.periods = periods_tmp;
+    }
+  }
+
+  getDateWithoutHours(date: Date) {
+    return new Date(date.getFullYear(), date.getMonth(), date.getDate());
+  }
+
+  goToLeft(event) {
+    event.preventDefault();
+    const current_date = this.days_of_Week_titles[0].date;
+    current_date.setDate(current_date.getDate() - 1);
+    this.days_of_Week_titles = this.startAndEndDaysOfWeek(current_date);
+    this.refreshDaysWeek();
+  }
+
+  goToRight(event) {
+    event.preventDefault();
+    const current_date = this.days_of_Week_titles[6].date;
+    current_date.setDate(current_date.getDate() + 1);
+    this.days_of_Week_titles = this.startAndEndDaysOfWeek(current_date);
+    this.refreshDaysWeek();
```

Comment:
That's a big copy/paste, please do a generic function and create two other to call with the good parameter